datatypes/src/raster/grid.rs

```diff
@@ -0,0 +1,416 @@
+use std::{
+    fmt::Debug,
+    ops::{Add, Div, Mul, Rem, Sub},
+};
+
+use num_traits::{One, Zero};
+use serde::{Deserialize, Serialize};
+
+use crate::util::Result;
+
+use super::grid_bounds::GridBoundingBox;
+
+pub trait GridSize {
+    /// An array with one entry for each axis of the `GridRect`
+    type ShapeArray: AsRef<[usize]>;
+
+    /// The number of axis of the `GridRect`
+    const NDIM: usize;
+
+    // size per axis in [.., y, x] order
+    fn axis_size(&self) -> Self::ShapeArray;
+    /// Size of the x-axis
+    fn axis_size_x(&self) -> usize {
+        match *self.axis_size().as_ref() {
+            [] => 0,
+            [.., a] => a,
+        }
+    }
+
+    /// Size of the y-axis
+    fn axis_size_y(&self) -> usize {
+        match *self.axis_size().as_ref() {
+            [] => 0,
+            [_] => 1,
+            [.., b, _] => b,
+        }
+    }
+
+    /// The number of elements the `GridRect` contains. This is bounded by size(usize).
```

Comment:
there is nor GridRect anymore. Also: what does the bound mean?

Reply:
reworked the comments.